src/main/java/gregtech/api/capability/impl/miner/MinerLogic.java

```diff
@@ -164,22 +164,24 @@ public void performMining() {
             NonNullList<ItemStack> blockDrops = NonNullList.create();
             IBlockState blockState = metaTileEntity.getWorld().getBlockState(blocksToMine.getFirst());
 
-            // if the block is not air or cobblestone., harvest it
+            // check to make sure the ore is still there,
+            while(!GTUtility.isOre(GTUtility.toItem(blockState))) {
+                blocksToMine.removeFirst();
+                if (blocksToMine.isEmpty()) break;
+                blockState = metaTileEntity.getWorld().getBlockState(blocksToMine.getFirst());
+            }
+            // When we are here we have an ore to mine! I'm glad we aren't threaded
             if (GTUtility.isOre(GTUtility.toItem(blockState))) {
```

Comment:
This should probably get an added check for `!blocksToMine.isEmpty()`

Reply:
Agreed, whilst it shouldn't change anything, it makes that line much more obvious in its checking.